db/crud.go

```diff
@@ -869,6 +869,33 @@ func (db *DatabaseCollectionWithUser) OnDemandImportForWrite(ctx context.Context
 	return nil
 }
 
+// updateHLV updates the HLV in the sync data appropriately based on what type of document update event we are encountering
+func (db *DatabaseCollectionWithUser) updateHLV(d *Document, docUpdateEvent uint32) (*Document, error) {
+
+	if d.HLV == nil {
+		d.HLV = &HybridLogicalVector{}
+	}
+	switch docUpdateEvent {
+	case BlipWriteEvent:
```

Comment:
I don't think BlipWriteEvent and SGWriteEvent are the best descriptions for these options.  Today the REST API has the ability to invoke PutExistingRev, and that might be something we end up supporting for versions as well.  Given that I'm thinking it would be preferable to define these based on the functional case.  

I also think these are more like types of updates, as opposed to 'events'.  I'm open to suggestions - maybe something like `docUpdateType` for the type, with `Import`, `ExistingVersion` and `NewVersion`?